cmake/HPX_SetupHIP.cmake

```diff
@@ -0,0 +1,50 @@
+# Copyright (c)      2020 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+# No need for the user to specify the option explicitly
+if("${CMAKE_CXX_COMPILER}" MATCHES "hipcc$")
+  include(HPX_Option)
+  hpx_option(
+    HPX_WITH_HIP BOOL "Enable compilation with HIPCC (default: ON)" ON ADVANCED
+  )
+else()
+  hpx_option(
+    HPX_WITH_HIP BOOL "Enable compilation with HIPCC (default: OFF)" OFF
+    ADVANCED
+  )
+endif()
+
+if(HPX_WITH_HIP)
+
+  if(HPX_WITH_CUDA)
+    hpx_error(
+      "Both HPX_WITH_CUDA and HPX_WITH_HIP are ON. Please choose one of \
+    them for HPX to work properly"
+    )
+  endif(HPX_WITH_CUDA)
+
+  # Needed on rostam
+  list(APPEND CMAKE_PREFIX_PATH $ENV{HIP_PATH}/lib/cmake/hip)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/AMDDeviceLibs)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/amd_comgr)
+  list(APPEND CMAKE_PREFIX_PATH $ENV{DEVICE_LIB_PATH}/cmake/hsa-runtime64)
+  # Setup hipblas (creates roc::hipblas)
+  find_package(hipblas HINTS $ENV{HIPBLAS_ROOT} CONFIG)
+  if(NOT hipblas_FOUND)
+    hpx_error(
+      "Hipblas could not be found, please specify HIPBLAS_ROOT to point to the \
+      correct location"
+    )
+  endif()
+  target_include_directories(
+    roc::hipblas SYSTEM INTERFACE ${hipblas_INCLUDE_DIRS}
```

Comment:
I remember having missing symbols but I can try it again cause I did a lot of changes since then

Reply:
Missing symbols would be because of a missing shared library, not missing include directory. Did you really have missing symbols or missing header files?